packages/remix-react/data.ts

```diff
@@ -40,12 +40,23 @@ export async function fetchData(
     init.method = request.method;
 
     let contentType = request.headers.get("Content-Type");
-    init.body =
-      // Check between word boundaries instead of startsWith() due to the last
-      // paragraph of https://httpwg.org/specs/rfc9110.html#field.content-type
-      contentType && /\bapplication\/x-www-form-urlencoded\b/.test(contentType)
-        ? new URLSearchParams(await request.text())
-        : await request.formData();
+
+    // Check between word boundaries instead of startsWith() due to the last
+    // paragraph of https://httpwg.org/specs/rfc9110.html#field.content-type
+    if (contentType && /\bapplication\/json\b/.test(contentType)) {
+      init.headers = { "Content-Type": contentType };
+      init.body = JSON.stringify(await request.json());
+    } else if (contentType && /\btext\/plain\b/.test(contentType)) {
+      init.headers = { "Content-Type": contentType };
+      init.body = await request.text();
+    } else if (
+      contentType &&
+      /\bapplication\/x-www-form-urlencoded\b/.test(contentType)
+    ) {
+      init.body = new URLSearchParams(await request.text());
+    } else {
+      init.body = await request.formData();
+    }
```

Comment:
Ensure we proxy the RR request over the wire to Remix with the correct content type and `body`